psi4/src/psi4/libmints/pseudospectral.h

```diff
@@ -95,7 +95,7 @@ class PseudospectralInt : public OneBodyAOInt {
     void use_omega(bool yes) { use_omega_ = yes; }
 
     /// Does the method provide first derivatives?
-    bool has_deriv1() { return false; }
+    bool has_deriv1() override { return false; }
```

Comment:
Does it make sense to have a default implementation for the parent class and also to demand (I think, through `override`) that the derived implement it?

Reply:
Having these virtual methods with implementation in the base class means that all subclasses will conform to the same interface. For the particular case of `has_deriv1` and `has_deriv2` only subclasses that have 1st and 2nd derivative must override the default implementation (so that it returns `true`). For all other subclasses there might be a slight performance penalty.

Reply:
got it. `override` demands derived have same interface, not that derived reimplement.

Reply:
That's a demand coming from `virtual` in the base class, actually. `override` is there to declare that the method in the derived class is overriding the one in the base class, such that the compiler can actually check we're doing it correctly.

Reply:
Just repeating stuff i'm sure, but my understanding is that it's to avoid something like the following.  If we have some function in the class Base `virtual Base::func(float arg)`.  If there's a class Derived that derives from Base with a function `Derived::func(int arg)` it could be something the programmer intended to implement to overload the base class but accidentally got the wrong arg type; it would compile (probably with warnings) because the int can be converted to a float.  The problem is that a `Base*` pointer would then call the version from Base, while a `Derived*` pointer would call the version from Derived because `Derived::func` doesn't overload `Base::func` -- it's a different function -- when both pointers should result in a call to `Derived::func`.  The `overload` descriptor just guarantees that similarly named functions have the same interface to avoid this confusion.